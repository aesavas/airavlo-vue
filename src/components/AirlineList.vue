<script setup lang="ts">
defineProps({
  majorAirlines: Array<string>
})
</script>

<template>
  <div
    class="accordion mt-2 mb-3"
    id="accordion"
  >
    <div class="accordion-item">
      <h2 class="accordion-header">
        <button
          class="accordion-button collapsed"
          type="button"
          data-bs-toggle="collapse"
          data-bs-target="#flush-collapseOne"
          aria-expanded="false"
          aria-controls="flush-collapseOne"
        >
          <p><strong>Major Airlines:</strong></p>
        </button>
      </h2>
      <div
        id="flush-collapseOne"
        class="accordion-collapse collapse"
        data-bs-parent="#accordion"
      >
        <div class="accordion-body">
          <div
            v-for="value in majorAirlines"
            :key="value"
          >ðŸ›« {{ value }}</div>
        </div>
      </div>
    </div>
  </div>
</template>
