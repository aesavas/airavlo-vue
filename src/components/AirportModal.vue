<script setup lang="ts">
import type { IAirport } from '@/types'
import AirlineList from './AirlineList.vue'

interface Props {
  airport: IAirport | null
  show?: boolean
}

withDefaults(defineProps<Props>(), {
  show: false,
  airport: null
})

defineEmits(['close'])
</script>

<template>
  <!-- Main modal container -->
  <div
    class="modal fade"
    :class="{ 'show': show }"
    :style="{ 'display': show ? 'block' : 'none' }"
    tabindex="-1"
  >
    <div class="modal-dialog modal-dialog-centered modal-dialog-scrollable">
      <div
        v-if="airport"
        class="modal-content"
      >
        <div class="modal-header">
          <h5 class="modal-title">{{ airport.name }} ({{ airport.code }})</h5>
          <button
            type="button"
            class="btn-close"
            @click="$emit('close')"
          ></button>
        </div>
        <div class="modal-body">
          <div class="row">
            <div class="col-6">
              <p><strong>City:</strong> {{ airport.city }}</p>
              <p><strong>Country:</strong> {{ airport.country }}</p>
              <p><strong>Continent:</strong> {{ airport.continent }}</p>
            </div>
            <div class="col-6">
              <p><strong>Year Opened:</strong> {{ airport.yearOpened }}</p>
              <p><strong>Type:</strong> {{ airport.type }}</p>
              <p><strong>Passenger Volume:</strong> {{ airport.passengerVolume }}</p>
            </div>
          </div>
          <div class="row">
            <p><strong>Website:</strong> <a :href="airport.website">{{ airport.website }}</a></p>
          </div>
          <div class="row">
            <AirlineList :major-airlines="airport.majorAirlines" />
          </div>
          <div class="row">
            <iframe
              width="300"
              height="200"
              frameborder="0"
              style="border:0"
              :src="`https://www.google.com/maps?q=${airport.coordinates.lat},${airport.coordinates.lng}&z=15&output=embed`"
              allowfullscreen
            >
            </iframe>
          </div>
        </div>
      </div>
    </div>
  </div>
  <!-- Backdrop element required by Bootstrap -->
  <div
    v-if="show"
    class="modal-backdrop fade show"
  ></div>
</template>
