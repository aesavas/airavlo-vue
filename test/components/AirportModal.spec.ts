import AirlineList from '@/components/AirlineList.vue'
import AirportModal from '@/components/AirportModal.vue'
import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'

describe("Airport Modal Component", () => {
  it("render airport modal test with show prop false", () => {
    const wrapper = mount(AirportModal, {
      props: {
        show: false,
        airport: null
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("render airport modal with props", () => {
    const wrapper = mount(AirportModal, {
      props: {
        show: true,
        airport: {
          name: "Test Airport",
          code: "TEST",
          city: "Test City",
          country: "Test Country",
          continent: "Test Continent",
          yearOpened: 2020,
          type: "Test Type",
          timezone: "+03:00",
          passengerVolume: "Major Hub",
          website: "https://testairport.com",
          majorAirlines: ["Test Airline 1", "Test Airline 2"],
          coordinates: {
            lat: 0,
            lng: 0
          }
        }
      }
    })

    expect(wrapper.findComponent(AirlineList).exists()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
