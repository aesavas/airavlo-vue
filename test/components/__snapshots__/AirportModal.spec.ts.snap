// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Airport Modal Component > render airport modal test with show prop false 1`] = `
"<!-- Main modal container -->
<div class="modal fade" style="display: none;" tabindex="-1">
  <div class="modal-dialog modal-dialog-centered modal-dialog-scrollable">
    <!--v-if-->
  </div>
</div>
<!-- Backdrop element required by Bootstrap -->
<!--v-if-->"
`;

exports[`Airport Modal Component > render airport modal with props 1`] = `
"<!-- Main modal container -->
<div class="modal fade show" style="display: block;" tabindex="-1">
  <div class="modal-dialog modal-dialog-centered modal-dialog-scrollable">
    <div class="modal-content">
      <div class="modal-header">
        <h5 class="modal-title">Test Airport (TEST)</h5><button type="button" class="btn-close"></button>
      </div>
      <div class="modal-body">
        <div class="row">
          <div class="col-6">
            <p><strong>City:</strong> Test City</p>
            <p><strong>Country:</strong> Test Country</p>
            <p><strong>Continent:</strong> Test Continent</p>
          </div>
          <div class="col-6">
            <p><strong>Year Opened:</strong> 2020</p>
            <p><strong>Type:</strong> Test Type</p>
            <p><strong>Passenger Volume:</strong> Major Hub</p>
          </div>
        </div>
        <div class="row">
          <p><strong>Website:</strong> <a href="https://testairport.com">https://testairport.com</a></p>
        </div>
        <div class="row">
          <div class="accordion mt-2 mb-3" id="accordion">
            <div class="accordion-item">
              <h2 class="accordion-header"><button class="accordion-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseOne" aria-expanded="false" aria-controls="flush-collapseOne">
                  <p><strong>Major Airlines:</strong></p>
                </button></h2>
              <div id="flush-collapseOne" class="accordion-collapse collapse" data-bs-parent="#accordion">
                <div class="accordion-body">
                  <div>ðŸ›« Test Airline 1</div>
                  <div>ðŸ›« Test Airline 2</div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="row"><iframe width="300" height="200" frameborder="0" style="border: 0px;" src="https://www.google.com/maps?q=0,0&amp;z=15&amp;output=embed" allowfullscreen=""></iframe></div>
      </div>
    </div>
  </div>
</div>
<!-- Backdrop element required by Bootstrap -->
<div class="modal-backdrop fade show"></div>"
`;
